### PR TITLE
improve domain-settings, tweak avatar-mixer sending

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: d602c713466feea6a0374c32e8c786b1a29f7e4b available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b32aecc4568723e6c91c97264617c10e6ba866e5 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d0e6129239a1aa29345b96a3fbd50b9161bfb02 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7b4522e95024b3ac01cc692cd7e12509931e7451 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: edade7baba54d381ed0f99533a42768cdcbdc5d1 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7ce62f49e284ffcf8ed74c933d4380ab9d9b16e6 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b9c9ff6a1454b3d11683ec27366e0733dcfd5163 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 19a95cb25a2e059761d4dc013ab7297c4e499c84 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 455b91c2a5069aab3631e8116bbd181bacd62f88 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0dbc4780ab48cf25c73ba2cbfae4ec137b46dcdf available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6cff75f7c14cdef57b7932a501726e57811237c9 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

File: interface/src/avatar/Avatar.cpp [hunk 1]
```diff
@@ -715,19 +715,18 @@ void Avatar::renderDisplayName() {
     QString renderedDisplayName = _displayName;
     
     if (DependencyManager::get<AvatarManager>()->shouldShowReceiveStats()) {
-        const float BYTES_PER_KILOBYTE = 1000.0f;
-        float kilobytesPerSecond = getAverageBytesReceivedPerSecond() / BYTES_PER_KILOBYTE;
+        const float KILOBITS_PER_BYTE = 125.0f;
```
Comment (sethalves) [inline interface/src/avatar/Avatar.cpp]: fix name  KILOBITS_PER_BYTE

File: libraries/networking/src/JSONBreakableMarshal.cpp [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+            // use a regex to look for surrounding quotes first
+            const QString JSON_STRING_REGEX = "^\"([\\s\\S]*)\"$";
+            QRegExp stringRegex(JSON_STRING_REGEX);
+             
+            if (stringRegex.indexIn(marshalValue) != -1) {
+                // set the result to the string value
+                result = stringRegex.cap(1);
+            } else {
+                // we failed to convert the value to anything, set the result to our unknown value 
+                result = JSON_UNKNOWN_AS_STRING;
```
Comment (sethalves) [inline libraries/networking/src/JSONBreakableMarshal.cpp]: maybe a qDebug here?

File: libraries/shared/src/SimpleMovingAverage.cpp [hunk 3]
```diff
@@ -52,7 +52,7 @@ void SimpleMovingAverage::reset() {
 
 float SimpleMovingAverage::getEventDeltaAverage() const {
     return (ONE_MINUS_WEIGHTING * _eventDeltaAverage) +
-        (WEIGHTING * ((usecTimestampNow() - _lastEventTimestamp) / 1000000.0f));
+        (WEIGHTING * ((usecTimestampNow() - _lastEventTimestamp) / 1000000.0f ));
```
Comment (sethalves) [inline libraries/shared/src/SimpleMovingAverage.cpp]: why the stray space?

**Comment by hifi-gustavo** [comment]
Build console output for commit: aeb04cf9c319f7179bd5f90fed163af4c37b5d70 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

File: tests/shared/src/MovingMinMaxAvgTests.cpp [hunk 4]
```diff
@@ -59,7 +59,7 @@ void MovingMinMaxAvgTests::runAllTests() {
 
             assert(stats.getMin() == min);
             assert(stats.getMax() == max);
-            assert(abs(stats.getAverage() / average - 1.0) < 0.000001 || abs(stats.getAverage() - average) < 0.000001);
+            assert(fabs(stats.getAverage() / average - 1.0) < 0.000001 || fabs(stats.getAverage() - average) < 0.000001);
```
Comment (sethalves) [inline tests/shared/src/MovingMinMaxAvgTests.cpp]: use epsilon constant?

File: tests/shared/src/MovingMinMaxAvgTests.cpp [hunk 5]
```diff
@@ -78,7 +78,7 @@ void MovingMinMaxAvgTests::runAllTests() {
 
                 assert(stats.getWindowMin() == windowMin);
                 assert(stats.getWindowMax() == windowMax);
-                assert(abs(stats.getAverage() / average - 1.0) < 0.000001 || abs(stats.getAverage() - average) < 0.000001);
+                assert(fabs(stats.getAverage() / average - 1.0) < 0.000001 || fabs(stats.getAverage() - average) < 0.000001);
```
Comment (sethalves) [inline tests/shared/src/MovingMinMaxAvgTests.cpp]: epsilon?

**Comment by hifi-gustavo** [comment]
Build console output for commit: fb7646b02b6a69832d34a72b7fbf4d23ed3415d5 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 87e586dfe6a30a174eb35e76c7979f70591c99ff available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 14fa89858e74c44793eca1c48f4b7eaea83df650 available at: https://gist.github.com/9b4a78283885711caada<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4747.dmg

**Pull request merged**